src/coins.h

```diff
@@ -301,8 +301,13 @@ class CCoinsViewCache : public CCoinsViewBacked
 
     /**
      * Return a reference to Coin in the cache, or a pruned one if not found. This is
-     * more efficient than GetCoin. Modifications to other cache entries are
-     * allowed while accessing the returned pointer.
+     * more efficient than GetCoin.
+     *
+     * Generally, do not hold the reference returned for more than a short scope.
+     * While the current implementation allows for modifications to the contents
+     * of the cache while holding the reference, this behavior should not be relied
+     * on! To be safe, best to not hold the returned reference through any other
+     * calls to this cache.
```

Comment:
Due to the multi-threaded nature of the BU code, its dangerous to use this at all, which is why it lack of locking is indicated by prepending the function with an _.  In BU, the "CoinAccessor" wrapper should be used since it properly handles locking over the duration of your use of the coin.  (look at your change in tx_verify.cpp to see a use a CoinAccessor)  We should change this comment to reflect BU's greater strictness requirement.

Reply:
fair enough, will do.

Reply:
done